examples/Main.hs

```diff
@@ -48,6 +56,53 @@ example2 = plam $ \x -> pmatch x $ \case
   PLeft n -> n + 1
   PRight n -> n - 1
 
+data SampleRecord f = SampleRecord
```

Comment:
We probably want a `examples/Examples/LetRec.hs` file or similar. We've been putting too many things in this one module.